src/ai/backend/common/bgtask.py

```diff
@@ -92,12 +104,12 @@ async def _pipe_builder(r: Redis) -> Pipeline:
 class BackgroundTaskManager:
     event_producer: EventProducer
     ongoing_tasks: weakref.WeakSet[asyncio.Task]
-    task_update_queues: Set[asyncio.Queue[Sentinel | BgtaskEvents]]
+    task_update_queues: DefaultDict[uuid.UUID, Set[asyncio.Queue[Sentinel | BgtaskEvents]]]
 
     def __init__(self, event_producer: EventProducer) -> None:
         self.event_producer = event_producer
         self.ongoing_tasks = weakref.WeakSet()
-        self.task_update_queues = set()
+        self.task_update_queues = defaultdict(set)
```

Comment:
I can't see the expression to create fresh set at desired task_id key.

Reply:
defaultdict will create the fresh set for the given task_id key if given key doesn't exist.

Reply:
Oh that's awesome.